core/src/test/scala/org/apache/spark/SparkContextSchedulerCreationSuite.scala

```diff
@@ -159,7 +159,7 @@ class SparkContextSchedulerCreationSuite
       assert(sched.backend.getClass === expectedClass)
     } catch {
       case e: UnsatisfiedLinkError =>
-        assert(e.getMessage.contains("no mesos in"))
+        assert(e.getMessage.contains("no mesos in") || e.getMessage.contains("mesos (Not found in java.library.path)"))
```

Comment:
So, if it's definitely an `UnsatisfiedLinkError`, I imagine it's sufficient to look for just "mesos"? that would differentiate it from not finding other libs without having to depend on the text like this. Would that work?